source/patterns/@aws-solutions-constructs/aws-apigatewayv2websocket-sqs/lib/index.ts

```diff
@@ -0,0 +1,178 @@
+   *
+   * @default - None
+   */
+  readonly existingWebSocketApi?: apigwv2.WebSocketApi;
+
+  /**
+   * Optional user-provided props to override the default props for the API Gateway.
+   *
+   * @default - Default properties are used.
+   */
+  readonly webSocketApiProps?: apigwv2.WebSocketApiProps;
+
+  /**
+   * User provided props to override the default props for the SQS queue.
+   *
+   * @default - Default props are used
+   */
+  readonly queueProps?: sqs.QueueProps;
+
+  /**
+   * Existing instance of SQS queue object, providing both this  and queueProps will cause an error
+   */
+  readonly existingQueueObj?: sqs.Queue;
+
+  /**
+   * Optional user-provided props to override the default props for the log group.
+   *
+   * @default - Default props are used
+   */
+  readonly logGroupProps?: logs.LogGroupProps;
+  /**
+   * If no key is provided, this flag determines whether the queue is encrypted with a new CMK or an AWS managed key.
+   * This flag is ignored if any of the following are defined: queueProps.encryptionMasterKey, encryptionKey or encryptionKeyProps.
+   *
+   * @default - False if queueProps.encryptionMasterKey, encryptionKey, and encryptionKeyProps are all undefined.
+   */
+  readonly enableEncryptionWithCustomerManagedKey?: boolean;
+  /**
+   * An optional, imported encryption key to encrypt the SQS Queue with.
+   *
+   * @default - None
+   */
+  readonly encryptionKey?: kms.Key;
+  /**
+   * Optional user provided properties to override the default properties for the KMS encryption key used to encrypt the SQS queue with.
+   *
+   * @default - None
+   */
+  readonly encryptionKeyProps?: kms.KeyProps;
+  /**
+   * Whether to deploy a secondary queue to be used as a dead letter queue.
+   *
+   * @default - required field.
+   */
+  readonly deployDeadLetterQueue?: boolean;
+  /**
+   * Optional user provided properties for the dead letter queue
+   *
+   * @default - Default props are used
+   */
+  readonly deadLetterQueueProps?: sqs.QueueProps;
+  /**
+   * The number of times a message can be unsuccessfully dequeued before being moved to the dead-letter queue.
+   *
+   * @default - required only if deployDeadLetterQueue = true.
+   */
+  readonly maxReceiveCount?: number;
+  /**
+   * API Gateway Request Template for the default route.
+   *
+   * @default - "Action=ReceiveMessage"
+   */
+  readonly defaultRouteRequestTemplate?: { [contentType: string]: string };
+
+  /**
+   * Whether to create a default route. If set to true, then it will use the value supplied with `defaultRouteRequestTemplate`
+   *
+   * @default - false.
+   */
+  readonly createDefaultRoute?: boolean;
+
+  /**
+   * User provided websocket authorizer. If not provided, the constructs defaults to
+   * {@link https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigatewayv2.WebSocketAuthorizer.html WebSocketAuthorizer}
```

Comment:
should we update this comment to also mention if the default WebSocketAuthorizer is configured with IAM or LAMBDA authorizer type? As we create this by default, my concern would be customers would be left with no way to access to the API without doing some additional investigation into what was deployed.

Reply:
The authorizer is no longer required. An authorizer is only allowed with `$connect`. The construct does not create `$connect`. Hence this property is not required. Removing it.